File: 19-remove-nth-node-from-end-of-list/remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def removeNthFromEnd(self, head, n):
        """
        :type head: ListNode
        :type n: int
        :rtype: ListNode
        """
        l=0
        x=head
        while x:
            l+=1
            x=x.next
        n=l-n+1
        count=2
        if n==1:
            x=head
            head=x.next
            x.next=None
            return head
        x=head
        while x:
            if count==n:
                t=x.next
                x.next=t.next
                t.next=None
            x=x.next
            count+=1
        
        return head

        
        
        


        

File: 19-remove-nth-node-from-end-of-list/remove-nth-node-from-end-of-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode* x = head; 
        int size = 0;
        while(x){
            x=x->next;
            size++;
        }
        ListNode*dummy = new ListNode;
        dummy->next  =  head;
        n=size-n;
        
        int i = 0;
        x=dummy;
        while(i++!=n){
            x=x->next;        
        }
        ListNode*temp =x->next;
        x->next = temp->next;
        temp->next = NULL;
        return dummy->next;
    }
};